mantidproject/mantid
# Indirect - Container Subtraction - Allow for workspaces in non-wavelength units

Currently, the Indirect Container Subtraction interface only accepts workspaces in units of wavelength. Output from the Elwin interface, as an example, is not in units of wavelength. This is an unnecessary check and is as a result of the interface using the ApplyPaalman pings algorithm to do the subtraction.

This PR updates the Container Subtraction interface to allow for container subtraction when the sample input and container aren't in units of wavelength.

**To test:**
1. Navigate to Interfaces > Indirect > Corrections > Container Subtraction.
2. Provide the supplied sample_elf.nxs workspace in the 'Sample' field.
3. Provide the supplied container_elf.nxs workspace in the 'Container' field.
4. Modify the shift and scale values in the respective spinners, and ensure the preview plot updates correctly.
5. Click 'Run'.
6. Ensure an output Subtracted workspace is created and the Sample (black), Container (red) and Subtracted (blue) plots are added to the mini-plot.

**Test Files:**
[elwin_container_subtraction_files.zip](https://github.com/mantidproject/mantid/files/1407214/elwin_container_subtraction_files.zip)



Fixes #19507. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
